### PR TITLE
Adds a subsystem for ai movement

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
refactor: Adds a subsystem for ai movement
/:cl:

lets movement happen on its own subsystem, and moves it into datums so we can have different movement behaviors like JPS

### CODE REVIEW

**Comment by Qustinnus** [comment]
@ZeWaka why u laughin

**Comment by Qustinnus** [comment]
@ZeWaka reply right now

**Comment by ZeWaka** [comment]
> @ZeWaka reply right now
> @ZeWaka why u laughin

it's about time you caught up to goon

**Comment by Qustinnus** [comment]
fuck you i will shit in your cereal

**Comment by ZeWaka** [comment]
hey your code doesn't compile

**Comment by PKPenguin321** [comment]
qustinnus.... stop making us look bad......

**Comment by LemonInTheDark** [comment]
God damn it floyd you stole my idea, I want royalties

**Comment by Qustinnus** [comment]
yeah but then you'd leave it in review hell for 2 months ;^)

File: code/controllers/subsystem/processing/ai_movement.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@ PROCESSING_SUBSYSTEM_DEF(ai_movement)
 	priority = FIRE_PRIORITY_NPC_MOVEMENT
 	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
 	init_order = INIT_ORDER_AI_MOVEMENT
-	wait = 0.1 SECONDS
+	wait = 1
```
Comment (ZeWaka) [inline code/controllers/subsystem/processing/ai_movement.dm]: why would you do this, this is blasphemy

Comment (Qustinnus) [reply code/controllers/subsystem/processing/ai_movement.dm]: sugondese

**Pull request merged**